holoviews/plotting/widgets/__init__.py

```diff
@@ -108,7 +108,11 @@ def __init__(self, plot, renderer=None, **params):
         super(NdWidget, self).__init__(**params)
         self.id = plot.comm.target if plot.comm else uuid.uuid4().hex
         self.plot = plot
-        self.dimensions, self.keys = drop_streams(plot.streams,
```

Comment:
Right in these cases it just needs to know if a kdim has a corresponding stream or not, there is no actual clash at the DynamicMap level because each level of wrapping resolves it's own streams.